internal/provider/resource_gitlab_branch.go

```diff
@@ -0,0 +1,236 @@
+package provider
+
+import (
+	"context"
+	"log"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	gitlab "github.com/xanzy/go-gitlab"
+)
+
+var _ = registerResource("gitlab_branch", func() *schema.Resource {
+	return &schema.Resource{
+		Description: "This resource allows you to create and manage GitLab branches.",
+
+		CreateContext: resourceGitlabBranchCreate,
+		ReadContext:   resourceGitlabBranchRead,
+		DeleteContext: resourceGitlabBranchDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+		Schema: map[string]*schema.Schema{
```

Comment:
The schema is missing the fields `developers_can_merge` and `developers_can_push`.